hack/.shellcheck_failures

```diff
@@ -10,3 +10,12 @@
 ./cluster/gce/util.sh
 ./cluster/log-dump/log-dump.sh
 ./cluster/pre-existing/util.sh
+./test/cmd/apply.sh
```

Comment:
Why are we adding things? Can we first merge PRs to fix these issues? Ideally we get the whitelist down to zero.

Reply:
Yup! Working on it now ref the comment from @BenTheElder ! So ill add the fixes into this PR.

Just doing some investigation on https://github.com/koalaman/shellcheck/wiki/SC2251, that may be masking some test failures. Updating the code as we speak.

We have a bunch of test code like this:

```bash
set -o errexit
[...]
! command_that_should_fail
[...]
```

This test will succeed no matter what the result of `command_that_should_fail` is.

Reply:
Or do you want me to create separate PRs for each fix?

Reply:
Either way. In this PR is great too.

Reply:
Updated this PR now, but splitted different stuff into different commits. :smile: